backend/app/services/rag/embedding/factory.py

```
@@ -108,6 +112,17 @@ def create_embedding_model_from_crd(
     model_id = env.get("model_id")
     custom_headers = env.get("custom_headers", {})
 
+    # Decrypt API key if present (handles both encrypted and plain keys)
+    if api_key:
+        try:
+            api_key = decrypt_api_key(api_key)
+        except Exception as e:
```

Ruff (0.14.13)

Do not catch blind exception: `Exception`

(BLE001)